src/components/App.js

```diff
@@ -30,9 +30,20 @@ const parseOptions = options => {
     .filter(str => str !== '')
 }
 
+/**
+ * Allows sharing URLs such as ?b=idle,loading,bad,ok
+ */
+let optionsFromQueryParams = ''
+if (typeof URLSearchParams !== 'undefined') {
+  const params = new URLSearchParams (document.location.search.substring(1));
+  optionsFromQueryParams = params.get('b') || ''
+}
```

Comment:
This is a great idea, but I'd like to make a few changes before we merge it, and I already have an idea of how to take this a step further.

1) I'd like `b` to be changed to something more descriptive. Took me a while to understand it was short for `booleans`. I think we should name it `options` instead. Make it very clear to the user.

2) I think `optionsFromQueryParams` should be changed in 2 ways. First, let's make it a function, and second, let's name it `getInitialOptions`.

The function is so this is only run on initial render, but also it allows us to rename it something more appropriate while hiding the detail that we are checking query params in the function. I think it can look like this:

```javascript
const getInitialOptions = () => {
  let result = ''
  if (typeof URLSearchParams !== 'undefined') {
    const params = new URLSearchParams(document.location.search.substring(1))
    result = params.get('options') || ''
  }
}
// ...

const [options, setOptions] = React.useState(getInitialOptions)
```

Once we land this, I think there are 2 things we can do to make this even better.

1) Add a `useEffect` that updates the queryParams as options updates so that the link is immediately shareable
2) Add a button to copy the shareable link to the clipboard (or something similar).

Anyways, good work, I'll try and add a quick thing to run prettier on this later.

Reply:
Alrighty, I've added some changes. I used `getOptionsFromQueryParams` rather than `getInitialOptions` as I found myself using it for other use cases as well.

The `useHistory` hook currently synchronizes the browser location with state. Location history is pushed every time a new option is added or if the the string has changed beyond recognition. Hitting the back and forward buttons will update state as well.

Copying the browsers url to clipboard is "stolen" from: http://girfahelp.blogspot.com/2019/06/copy-current-url-to-clipboard-using.html